pkg/print/list/base.go

```diff
@@ -149,26 +54,20 @@ func (sc *StatusCollector) LatestStatus() map[object.ObjMetadata]event.StatusEve
 //nolint:gocyclo
 func (b *BaseListPrinter) Print(ch <-chan event.Event, previewStrategy common.DryRunStrategy, printStatus bool) error {
 	var actionGroups []event.ActionGroup
-	applyStats := &ApplyStats{}
-	pruneStats := &PruneStats{}
-	deleteStats := &DeleteStats{}
-	waitStats := &WaitStats{}
+	var statsCollector stats.Stats
 	statusCollector := &StatusCollector{
 		latestStatus: make(map[object.ObjMetadata]event.StatusEvent),
 	}
 	formatter := b.FormatterFactory(previewStrategy)
 	for e := range ch {
+		statsCollector.Handle(e)
```

Comment:
Why collect these if we're not gonna return them to the caller?
Doesn't that mean the user has to wrap their own and collect twice?

Reply:
Yeah, this means it will be collected twice if the users wrap it. 

So in this case we do need the stats in the printer itself, since we do print some of the stats as part of the output. We can't handle that from outside the printer without making significant changes to the output.

Reply:
So I think we have two options here:
* Make it the responsibility of the printers to collect stats and return it to the caller. This is useful for printers that needs to use the same stats anyway, but less useful for printers that doesn't (i.e. `table`). The change in https://github.com/kubernetes-sigs/cli-utils/pull/500 returns this in an error since that can be done without changing the `printer` interface, but we could potentially returns the stats as a separate return value.
* Make collecting stats separate from the printing process. The separation of concerns is somewhat compelling, but it does make for some double work if the implementation of a printer itself requires the stats. This is currently the case for the `json` and the `events` printer. We could revisit the output format such that the summaries are only printed at the end, and thus could be handled outside of the print code itself. But other printer implementations might still want to include information that requires stats.

I think I prefer the latter, as I agree that collecting the stats doesn't seem like something that should be the responsibility of the printer. I don't think collecting it twice would be a significant overhead.